### ISSUE TITLE
Date plus slow performance

### ISSUE THREAD

**Issue by PaulAtST** [moment/luxon#1130]
**Describe the bug**
The plus operator is very slow compared to the built in JS date methods, which is especially noticeable when making date series

**To Reproduce**
JS Date: 
```
let startDate = new Date(2020, 0, 0)
let endDate = new Date(2022, 0, 0)
let intervals = []

let date = new Date(startDate)

while (date.getTime() <= endDate.getTime()) {
    intervals.push(date)
    let oldDate = date
    date = new Date(oldDate)
    date.setHours(oldDate.getHours() + 1)
}
```

luxon: 
```
let startDate = new Date(2020, 0, 0)
let endDate = new Date(2022, 0, 0)
let intervals = []

let date = DateTime.fromMillis(startDate.getTime()).setZone('America/New_York')

while (date.ts <= endDate.getTime()) {
    intervals.push(date.toJSDate())
    date = date.plus({hours: 1})
}
```

**Actual vs Expected behavior**
I would expect similar perforce between the two code snippets above, but instead lexicon is taking substantially longer. Running the above snippets in mocha, I am seeing on average 16ms for the JS Date version, and 600 + MS for the luxon version. 

**Desktop**
- OS: OSX 10.15
- Node: 16.13
- Luxon version: 2.3.0
- Tiimezone: America/New_York (Haven't tested others)

**Additional context**
I can understand if luxon was a bit slower that the native date implementation, but I was very surprised to see such a difference. Also adding days or months with luxon appears slower then the native date methods, from some ad hock testing, but I did not generate exact benchmarks. Please let me know if there is anything else that would be helpful to provide.

**Comment by icambron** [comment]
The difference is likely to be `.setZone('America/New_York')` which means that Luxon must use the relatively slow Intl API to do computations, instead of the faster native date. The JS Date one, which is nto zone aware, doesn't have to do that. If you remove the explicit zone management, are they closer?

I'm all for performance issues and possible improvements, but let's compare apples to apples.

**Comment by PaulAtST** [comment]
@icambron Fair point. I finally got a chance to do some digging and here is what I found: 

Running luxon with the following code: 

```
let startDate = new Date(2020, 0, 0)
let endDate = new Date(2022, 0, 0)
let intervals = []

let date = DateTime.fromMillis(startDate.getTime())

while (date.ts <= endDate.getTime()) {
    intervals.push(date.toJSDate())
    date = date.plus({hours: 1})
}
```

Gets me to about 155ms. Which is much better than the 600ms I was getting with a set timezone, but still a lot slower then the vanilla JS code which is running at around 16ms. 

To try and optimize the luxon code a bit more I created the duration only once: 
```
 let startDate = new Date(2020, 0, 0)
let endDate = new Date(2022, 0, 0)
let intervals = []
let dur = Duration.fromObject({hours: 1})
        
let date = DateTime.fromMillis(startDate.getTime())
        
while (date.ts <= endDate.getTime()) {
        intervals.push(date.toJSDate())
        date = date.plus(dur)
}
```
This got me down about another 10ms to about 145ms

--------------------------------------------------------------------------

Now this deviates a bit from the original, but since I do need the correct starting time for the time sequence taking the timezone into account, I made a combined solution where I use luxon to get the start time, and then increment the UTC time to create the date series: 

```
let startDate = new Date(2020, 0, 0)
let endDate = new Date(2022, 0, 0)
let intervals = []        

let startAdjusted = DateTime.fromMillis(startDate.getTime()).setZone('America/New_York')
startAdjusted = startAdjusted.set({
    hour: 0,
    minute: 0, 
    second: 0,
    millisecond: 0
})

let date = startAdjusted.toJSDate()

while (date.getTime() <= endDate.getTime()) {
    intervals.push(date)
    date = new Date(date.getTime() + (1000 * 60 * 60))
}
```
This gets me to about 20ms in total. I am thinking for generating a hour by hour date sequence this should work, since DST affects day plus ranges. Can you think of an edge case maybe I am missing?
